arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -563,6 +573,17 @@ Result visitAnalyzers(TRI_vocbase_t& vocbase,
         return resultVisitor(visitor, vocbase, slice);
       }
     }
+    TRI_IF_FAILURE("CheckDBWhenNotSingleShardAndFlagChange") {
+      if (vocbase.server()
+              .getFeature<arangodb::ClusterFeature>()
+              .forceOneShard()) {
+        return {TRI_ERROR_DEBUG,
+                "force-one-shard true and db is not single shard"};
+      } else {
+        return {TRI_ERROR_DEBUG,
```

Comment:
Why is this case the error? Both false and both true sounds good?

I understand it's for test but still looks strange

Reply:
I see, but my question about why do you need fail point for case when forceOneShard == isOneShard